packages/@aws-cdk/aws-apigateway/lib/model.ts

```diff
@@ -1,46 +1,153 @@
-    public readonly modelId = 'Empty';
+export interface ModelOptions {
+  /**
+   * The content type for the model.
+   * @default None
+   */
+  readonly contentType?: string;
+
+  /**
+   * A description that identifies this model..
+   * @default None
+   */
+  readonly description?: string;
+
+  /**
+   * A name for the model.
+   *
+   * Important
+   *  If you specify a name, you cannot perform updates that
+   *  require replacement of this resource. You can perform
+   *  updates that require no or some interruption. If you
+   *  must replace the resource, specify a new name.
+   *
+   * @default <auto> If you don't specify a name,
+   *  AWS CloudFormation generates a unique physical ID and
+   *  uses that ID for the model name. For more information,
+   *  see Name Type.
+   */
+  readonly modelName?: string;
+
+  /**
+   * The schema to use to transform data to one or more output formats.
+   * Specify null ({}) if you don't want to specify a schema.
+   */
+  readonly schema: jsonSchema.JsonSchema;
 }
 
-/**
- * Represents a reference to a REST API's Error model, which is available
- * as part of the model collection by default. This can be used for mapping
- * error JSON responses from an integration to a client, where a simple
- * generic message field is sufficient to map and return an error payload.
- *
- * Definition
- * {
- *   "$schema" : "http://json-schema.org/draft-04/schema#",
- *   "title" : "Error Schema",
- *   "type" : "object",
- *   "properties" : {
- *     "message" : { "type" : "string" }
- *   }
- * }
- */
-export class ErrorModel implements IModel {
-    public readonly modelId = 'Error';
+export interface ModelProps extends ModelOptions {
+  /**
+   * The rest API that this model is part of.
+   *
+   * The reason we need the RestApi object itself and not just the ID is because the model
+   * is being tracked by the top-level RestApi object for the purpose of calculating it's
+   * hash to determine the ID of the deployment. This allows us to automatically update
+   * the deployment when the model of the REST API changes.
+   */
+  readonly restApi: IRestApi;
 }
 
-// TODO: Implement Model, enabling management of custom models.
\ No newline at end of file
+export class Model extends Resource implements IModel {
+  /**
+   * Represents a reference to a REST API's Error model, which is available
+   * as part of the model collection by default. This can be used for mapping
+   * error JSON responses from an integration to a client, where a simple
+   * generic message field is sufficient to map and return an error payload.
+   *
+   * Definition
+   * {
+   *   "$schema" : "http://json-schema.org/draft-04/schema#",
+   *   "title" : "Error Schema",
+   *   "type" : "object",
+   *   "properties" : {
+   *     "message" : { "type" : "string" }
+   *   }
+   * }
+   */
+  public static readonly ERROR_MODEL: IModel = ({ modelId: "Error", stack: null, node: null } as unknown) as IModel;
```

Comment:
The unknown cast is not a good idea and will blow up in people’s faces. It’s ok if you want IModel not to extend IConstruct and exclude the linter rule.

Reply:
YES! That's what I had been looking for, found it now, fixed and simplifies the code.